### PR TITLE
🌱  Split out test modules from capi operator modules

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
**What this PR does / why we need it**:

This small change separates capi operator modules from test suite, allowing independent use and external imports. Removes core dependency on capi testing framework. Follows similar pattern to CAPI test suite

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Review by Fedosin — APPROVED** [review]
/approve

awesome! thank you for doing this.

**Review by furkatgofurov7 — COMMENTED** [review]
This looks good, apart couple of suggestions: 

- since we have separated now test modules into its own folder and we ship it with its own Go module, we should:
    -  add notes informing about this change properly 
    -  tag and version the new module and as part of the release. We would have to create an annotated tag for /test folder and push it every time we do a release from the repo, once this goes in. For example, if we're releasing v0.5.0, we should also tag test/v0.5.0
    - add a path to new go.mod/sum files in the Makefile [`modules`](https://github.com/kubernetes-sigs/cluster-api-operator/blob/72873f9dbe9f37a74e473a55b73fa5d49444b0a8/Makefile#L315-L319) target where we run gomod tidy in all folders where we have go.mod/sum files and [`verify-modules`](https://github.com/kubernetes-sigs/cluster-api-operator/blob/72873f9dbe9f37a74e473a55b73fa5d49444b0a8/Makefile#L273-L278) target
    - add it to Dockerfile in https://github.com/kubernetes-sigs/cluster-api-operator/blob/main/Dockerfile#L29-L31

**Review by alexander-demicev — APPROVED** [review]
/lgtm
Thanks!

**Comment by alexander-demicev** [comment]
I don't know if comments from @furkatgofurov7 are blocking so I put the PR on hold, feel free to remove it if needed

**Comment by Danil-Grigorev** [comment]
Everything fixed. @furkatgofurov7 I think nothing is pending

File: docs/release.md [hunk 1]
```diff
@@ -10,11 +10,18 @@ git push -u upstream release-0.1
 ```
 
 ```bash
+# Export the tag of the release to be cut, e.g.:
 export RELEASE_TAG=v0.1.1
 
+# Create tags locally
+# Warning: The test tag MUST NOT be an annotated tag.
 git tag -s -a ${RELEASE_TAG} -m ${RELEASE_TAG}
+git tag test/${RELEASE_TAG}
```
Comment (furkatgofurov7) [inline docs/release.md:19]: let's sign this tag with gpg signature as above?

**Review by furkatgofurov7 — COMMENTED** [review]
Last nit:

File: docs/release.md (hunk 1 above)
Comment (Danil-Grigorev) [reply docs/release.md:19]: This process is coming from CAPI release document. They explicitly don’t sign test package. I’m not sure why exactly, but I’d rather stick with existing process for now: https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/release/release-tasks.md#repeatedly-cut-a-release-1

Comment (furkatgofurov7) [reply docs/release.md:19]: I wonder why, though, but it is fine. Thanks!

/lgtm

**Pull request merged**